plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -82,39 +86,19 @@ So we use a horrible hack and match on `build . unpackFoldrCString#` to "undo" t
 rule.
 -}
 
-{- Note [Addr#]
-String literals usually have type `Addr#`. This is not very nice for us, since we certainly don't have
-those.
-
-However, they should only appear wrapped by an `unpackCString#` or similar. Except if the string literal
-gets lifted out by GHC! This is no problem for our compiler, we will just compile the literal as a binding
-and put in a reference... except that the binding would need to be of type `Addr#`.
-
-So we use another horrible hack and pretend that `Addr#` is `String`, since we treat `unpackCString#` as doing nothing.
--}
-
 compileLiteral
     :: CompilingDefault uni fun m
     => GHC.Literal -> m (PIRTerm uni fun)
 compileLiteral = \case
     -- Just accept any kind of number literal, we'll complain about types we don't support elsewhere
     (GHC.LitNumber _ i _) -> pure $ PIR.embed $ PLC.mkConstant () i
-    GHC.LitString bs      ->
-        -- Convert the bytestring into a core expression representing the list
-        -- of characters, then compile that!
-        -- Note that we do *not* convert this into a PLC string, but rather a list of characters,
-        -- since that is what other Haskell code will expect.
-        let
-            str = T.unpack $ TE.decodeUtf8 bs
-            charExprs = fmap GHC.mkCharExpr str
-            listExpr = GHC.mkListExpr GHC.charTy charExprs
-        in compileExpr listExpr
-    GHC.LitChar c      -> pure $ PIR.embed $ PLC.mkConstant () (T.singleton c)
-    GHC.LitFloat _     -> throwPlain $ UnsupportedError "Literal float"
-    GHC.LitDouble _    -> throwPlain $ UnsupportedError "Literal double"
-    GHC.LitLabel {}    -> throwPlain $ UnsupportedError "Literal label"
-    GHC.LitNullAddr    -> throwPlain $ UnsupportedError "Literal null"
-    GHC.LitRubbish     -> throwPlain $ UnsupportedError "Literal rubbish"
+    GHC.LitString _       -> throwPlain $ UnsupportedError "(Non-overloaded) literal string"
```

Comment:
is it possible to include a recommended solution/action in the error message?

Reply:
Let me see if it's possible to trigger it, first. It might not be.

Reply:
This actually prompted me to put some of the old code back in, so we're better at recognizing string literals, so you actually get the error for them instead of nonsense about `unpackCString#`